src/core/server/elasticsearch/integration_tests/client.test.ts

```diff
@@ -55,3 +63,67 @@ describe('elasticsearch clients', () => {
     expect(resp.headers!.warning).toMatch('system indices');
   });
 });
+
+function createFakeElasticsearchServer() {
+  const firstRequestReceived$ = new Subject<boolean>();
+
+  const requestHandler = jest.fn((req, res) => {
+    // Reply with a 200 and empty response by default (intentionally malformed response)
+    res.writeHead(200);
+    res.end();
+
+    firstRequestReceived$.next(true);
+  });
+
+  const server = http.createServer(requestHandler);
+  server.listen(esTestConfig.getPort());
+  return { server, requestHandler, firstRequestReceived$ };
+}
+
+describe('fake elasticsearch', () => {
+  let esServer: http.Server;
+  let requestHandler: jest.Mock;
+  let kibanaServer: Root;
+
+  beforeAll(async () => {
+    kibanaServer = createRootWithCorePlugins(
+      {
+        logging: { silent: false },
+        status: { allowAnonymous: true },
+      },
+      { verbose: true }
+    );
+    const fakeServer = await createFakeElasticsearchServer();
+    esServer = fakeServer.server;
+    requestHandler = fakeServer.requestHandler;
+
+    await kibanaServer.preboot();
+    await kibanaServer.setup();
+    kibanaServer.start();
+    await fakeServer.firstRequestReceived$.pipe(first()).toPromise(); // Waiting for the first ES request instead of the start to complete, because start can't never complete due to waiting for ES to be ready
+  });
+
+  afterAll(async () => {
+    await kibanaServer.shutdown();
+    await esServer.close();
+  });
+
+  test('should return unknown product when it cannot perform the Product check (503 response)', async () => {
+    expect(requestHandler).toHaveBeenCalledWith(
+      expect.objectContaining({
+        method: 'GET',
+        url: '/',
+      }),
+      expect.anything()
+    );
+
+    // eslint-disable-next-line
+    const resp = await supertest(kibanaServer['server']['http']['prebootServer']['server']!.listener) // Mind that we are using the prebootServer at this point because the migration gets hanging, while waiting for ES to be correct
+      .get('/api/status')
+      .expect(503);
```

Comment:
It's turned out you cannot test it locally in the `dev` mode since `basePathProxy` blocks all the calls to the `http` preboot server 😞  https://github.com/elastic/kibana/blob/909d39b4da234f7b1279ea50e3d35205ed2c9be1/packages/kbn-cli-dev-mode/src/dev_server.ts#L198

Reply:
Hum, this case is supposed to be handled by

https://github.com/elastic/kibana/blob/237256a4bad936b74c56dd9f718b57526109fa58/src/core/server/bootstrap.ts#L88-L91

Reply:
> Hum, this case is supposed to be handled by

@pgayvallet Right now, this condition is always `false` because no plugins call `holdSetupUntilResolved`.